src/atomate2/common/flows/electrode.py

```diff
@@ -0,0 +1,141 @@
+"""Flow for electrode analysis."""
+
+from __future__ import annotations
+
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from jobflow import Flow, Maker
+from pymatgen.analysis.structure_matcher import ElementComparator, StructureMatcher
+
+from atomate2.common.jobs.electrode import get_stable_inserted_structure
+
+if TYPE_CHECKING:
+    from pymatgen.alchemy import ElementLike
+    from pymatgen.core.structure import Structure
+    from pymatgen.io.vasp.outputs import VolumetricData
+
+logger = logging.getLogger(__name__)
+
+__author__ = "Jimmy Shen"
+__email__ = "jmmshn@gmail.com"
+
+
+@dataclass
+class ElectrodeInsertionMaker(Maker, ABC):
+    """Attempt ion insertion into a structure.
+
+    The basic unit for cation insertion is:
+        [get_stable_inserted_structure]:
+            (static) -> (chgcar analysis) ->
+            N x (relax) -> (return best structure)
+
+    The workflow is:
+        [relax structure]
+        [get_stable_inserted_structure]
+        [get_stable_inserted_structure]
+        [get_stable_inserted_structure]
+        ... until the insertion is no longer topotactic.
+
+    This workflow requires the users to provide the following functions:
+        self.get_charge_density(task_doc: TaskDoc):
+            Get the charge density of a TaskDoc output from a calculation.
+        self.update_static_maker():
+            Ensure that the static maker will store the desired data.
```

Comment:
Is there a paper or anything that documents this further?

Reply:
Yeah link is here, I'll add it to the PR.

https://www.nature.com/articles/s41524-020-00422-3